src/qibolab/couplers.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union
+
+from qibolab.channels import Channel
+from qibolab.qubits import Qubit, QubitId
+
+CouplerId = Union[str, int]
+"""Type for Coupler names."""
+
+
+@dataclass
+class Coupler:
+    """Representation of a physical coupler.
+
+    Coupler objects are instantiated by :class:`qibolab.platforms.platform.Platform`
+    but they are passed to instrument designs in order to play pulses.
+
+    Args:
+        name (int, str): Qubit number or name.
+
+        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
+            send flux pulses to the qubit.
+        Other characterization parameters for the coupler, loaded from the runcard.
+    """
+
+    name: CouplerId
+
+    # TODO: I think is not needed
+    frequency: int = 0
+
+    sweetspot: float = 0
+
+    flux: Optional[Channel] = None
+    qubits: Optional[Dict[QubitId, Qubit]] = field(default_factory=dict)
+
+    @property
+    def channels(self):
+        if self.flux is not None:
+            yield self.flux
```

Comment:
I got it from the `Qubit` I assumed this woulbe be a nice way to still have `channels` in the` Coupler` as well even if it's only the `flux` one. It can also be said about `characterization`.

Reply:
Ok, let's close for the time being, definitely not a priority.